### ISSUE TITLE
Cannot do anything with `dotnet fsi` in .NET Core 3.1 preview

### ISSUE THREAD

**Issue by cartermp** [dotnet/fsharp#7811]
On macOS Catalina.

Can't do basic arithmetic:

<img width="507" alt="image" src="https://user-images.githubusercontent.com/6309070/68078725-a9141500-fd99-11e9-89a9-656479687cb8.png">

And pretty much nothing else works either.

**Comment by baronfel** [comment]
Did a bit of testing on preview two this morning:

* Debian 9 with 3.1.100 preview 2 can execute the above interaction
* Mac OS Mojave with 3.1.100 preview 2 _cannot_ execute the above interaction.  I can't even declare simple bindings:

```
> let m = 10m;;


error FS0193: Could not load file or assembly '"The system type 'System.Runtime.CompilerServices.CompilerGeneratedAttribute' was required but no referenced system DLL contained this type", Culture=neutral, PublicKeyToken=null'. The system cannot find the file specified.
```

**Comment by KevinRansom** [comment]
I just installed the most recent builds of 3.1.100-preview2-014569, 3.1.100-Preview3 and 5.0 from the dotnet cli site:
all worked for me on.  I will update to catalina, and see how it goes.

**MacOS Mojave 10.14.6**
````
Welcome to .NET Core 3.1!
---------------------
SDK Version: 3.1.100-preview2-014569

Telemetry
---------
The .NET Core tools collect usage data in order to help us improve your experience. The data is anonymous. It is collected by Microsoft and shared with the community. You can opt-out of telemetry by setting the DOTNET_CLI_TELEMETRY_OPTOUT environment variable to '1' or 'true' using your favorite shell.

Read more about .NET Core CLI Tools telemetry: https://aka.ms/dotnet-cli-telemetry

----------------
Explore documentation: https://aka.ms/dotnet-docs
Report issues and find source on GitHub: https://github.com/dotnet/core
Find out what's new: https://aka.ms/dotnet-whats-new
Learn about the installed HTTPS developer cert: https://aka.ms/aspnet-core-https
Use 'dotnet --help' to see available commands or visit: https://aka.ms/dotnet-cli-docs
Write your first app: https://aka.ms/first-net-core-app
--------------------------------------------------------------------------------------

Microsoft (R) F# Interactive version 10.7.0.0 for F# 4.7
Copyright (c) Microsoft Corporation. All Rights Reserved.

For help type #help;;

> 2+2;;
val it : int = 4

> 
````
````
Kevins-MBP:~/temp/xps>dotnet fsi

Welcome to .NET Core 3.1!
---------------------
SDK Version: 3.1.100-preview3-014639

Telemetry
---------
The .NET Core tools collect usage data in order to help us improve your experience. The data is anonymous. It is collected by Microsoft and shared with the community. You can opt-out of telemetry by setting the DOTNET_CLI_TELEMETRY_OPTOUT environment variable to '1' or 'true' using your favorite shell.

Read more about .NET Core CLI Tools telemetry: https://aka.ms/dotnet-cli-telemetry

----------------
Explore documentation: https://aka.ms/dotnet-docs
Report issues and find source on GitHub: https://github.com/dotnet/core
Find out what's new: https://aka.ms/dotnet-whats-new
Learn about the installed HTTPS developer cert: https://aka.ms/aspnet-core-https
Use 'dotnet --help' to see available commands or visit: https://aka.ms/dotnet-cli-docs
Write your first app: https://aka.ms/first-net-core-app
--------------------------------------------------------------------------------------

Microsoft (R) F# Interactive version 10.7.0.0 for F# 4.7
Copyright (c) Microsoft Corporation. All Rights Reserved.

For help type #help;;

> 2+2;;
val it : int = 4

> cc
````
and ...
````
Kevins-MBP:~/temp/xps>dotnet fsi

Welcome to .NET Core 5.0!
---------------------
SDK Version: 5.0.100-alpha1-015580

Telemetry
---------
The .NET Core tools collect usage data in order to help us improve your experience. The data is anonymous. It is collected by Microsoft and shared with the community. You can opt-out of telemetry by setting the DOTNET_CLI_TELEMETRY_OPTOUT environment variable to '1' or 'true' using your favorite shell.

Read more about .NET Core CLI Tools telemetry: https://aka.ms/dotnet-cli-telemetry

----------------
Installed an ASP.NET Core HTTPS development certificate.
To trust the certificate run 'dotnet dev-certs https --trust' (Windows and macOS only).
Learn about HTTPS: https://aka.ms/dotnet-https
----------------
Write your first app: https://aka.ms/dotnet-hello-world
Find out what's new: https://aka.ms/dotnet-whats-new
Explore documentation: https://aka.ms/dotnet-docs
Report issues and find source on GitHub: https://github.com/dotnet/core
Use 'dotnet --help' to see available commands or visit: https://aka.ms/dotnet-cli
--------------------------------------------------------------------------------------

Microsoft (R) F# Interactive version 10.7.0.0 for F# 4.7
Copyright (c) Microsoft Corporation. All Rights Reserved.

For help type #help;;

> Failed to install ctrl-c handler - Ctrl-C handling will not be available. Error was:
	Could not load file or assembly 'Mono.Posix, Version=2.0.0.0, Culture=neutral, PublicKeyToken=0738eb9f132ed756'. The system cannot find the file specified.
- 2+2;;
val it : int = 4

> 
````

**Comment by KevinRansom** [comment]
With Catalina

````
Last login: Mon Nov  4 20:16:30 on console

The default interactive shell is now zsh.
To update your account to use zsh, please run `chsh -s /bin/zsh`.
For more details, please visit https://support.apple.com/kb/HT208050.
Kevins-MBP:~ kevinr$ dotnet fsi
-bash: dotnet: command not found
Kevins-MBP:~ kevinr$ dotnet fsi

Microsoft (R) F# Interactive version 10.7.0.0 for F# 4.7
Copyright (c) Microsoft Corporation. All Rights Reserved.

For help type #help;;

> 2+2;;
val it : int = 4

> 

````

**Comment by KevinRansom** [comment]
@cartermp , if this still fails on your macbook, could I take a look please mate.

**Comment by eiriktsarpalis** [comment]
Related to #7369

I'm seeing this on and off with preview builds.

**Comment by cartermp** [comment]
@KevinRansom this reproduces with the latest .NET Core 3 preview:

<img width="413" alt="image" src="https://user-images.githubusercontent.com/6309070/68892181-4135c680-06d7-11ea-9c3f-b35cf0877898.png">

**Comment by KevinRansom** [comment]
@cartermp I left my MacBook at home, I can try when I get home, if you are going to be in today, could I look at it on your macbook?

thanks

Kevin

**Comment by cartermp** [comment]
I'm not in the office today still due to illness. These are my OS specs:

<img width="294" alt="image" src="https://user-images.githubusercontent.com/6309070/68897600-b22eab80-06e2-11ea-9a95-da112846f7bb.png">

**Comment by KevinRansom** [comment]
Still works for me
@cartermp , I am going to need to look at your machine, see if I can figure out whats up.
System Software Overview:

````
  System Version:	macOS 10.15.1 (19B88)
  Kernel Version:	Darwin 19.0.0
  Boot Volume:	MacBook HD
  Boot Mode:	Normal
  Computer Name:	Kevin’s MacBook Pro
  User Name:	Kevin Ransom (kevinr)
  Secure Virtual Memory:	Enabled
  System Integrity Protection:	Enabled
  Time since boot:	9 days 20:41

````


````
Kevins-MBP:kevinransom kevinr$ dotnet

Usage: dotnet [options]
Usage: dotnet [path-to-application]

Options:
  -h|--help         Display help.
  --info            Display .NET Core information.
  --list-sdks       Display the installed SDKs.
  --list-runtimes   Display the installed runtimes.

path-to-application:
  The path to an application .dll file to execute.
Kevins-MBP:kevinransom kevinr$ dotnet --info
.NET Core SDK (reflecting any global.json):
 Version:   3.1.100-preview3-014639
 Commit:    c46588ae63

Runtime Environment:
 OS Name:     Mac OS X
 OS Version:  10.15
 OS Platform: Darwin
 RID:         osx.10.15-x64
 Base Path:   /usr/local/share/dotnet/sdk/3.1.100-preview3-014639/

Host (useful for support):
  Version: 3.1.0-preview3.19553.2
  Commit:  13f35c3d86

.NET Core SDKs installed:
  3.1.100-preview3-014639 [/usr/local/share/dotnet/sdk]

.NET Core runtimes installed:
  Microsoft.AspNetCore.App 3.1.0-preview3.19554.4 [/usr/local/share/dotnet/shared/Microsoft.AspNetCore.App]
  Microsoft.NETCore.App 3.1.0-preview3.19553.2 [/usr/local/share/dotnet/shared/Microsoft.NETCore.App]

To install additional .NET Core runtimes or SDKs:
  https://aka.ms/dotnet-download
Kevins-MBP:kevinransom kevinr$ dotnet fsi

Microsoft (R) F# Interactive version 10.7.0.0 for F# 4.7
Copyright (c) Microsoft Corporation. All Rights Reserved.

For help type #help;;

> 2+2;;
val it : int = 4

> 

````

**Comment by abelbraaksma** [comment]
@KevinRansom, just to note, the two versions of the preview are not identical. @cartermp uses 014645 and you:

> Version:   3.1.100-preview3-014639

Considering that the bug comes and goes, reprodibililty may have to do with that.

**Comment by KevinRansom** [comment]
For the sceptics among us here it is working on:
.NET Core SDKs installed:
  3.1.100-preview3-014645 [/usr/local/share/dotnet/sdk

I am going to need to check it out on a machine that fails, to discover what the environment issues are.

````
Kevins-MBP:~ kevinr$ dotnet --info
.NET Core SDK (reflecting any global.json):
 Version:   3.1.100-preview3-014645
 Commit:    b32d27f4b3

Runtime Environment:
 OS Name:     Mac OS X
 OS Version:  10.15
 OS Platform: Darwin
 RID:         osx.10.15-x64
 Base Path:   /usr/local/share/dotnet/sdk/3.1.100-preview3-014645/

Host (useful for support):
  Version: 3.1.0-preview3.19553.2
  Commit:  13f35c3d86

.NET Core SDKs installed:
  3.1.100-preview3-014645 [/usr/local/share/dotnet/sdk]

.NET Core runtimes installed:
  Microsoft.AspNetCore.App 3.1.0-preview3.19555.2 [/usr/local/share/dotnet/shared/Microsoft.AspNetCore.App]
  Microsoft.NETCore.App 3.1.0-preview3.19553.2 [/usr/local/share/dotnet/shared/Microsoft.NETCore.App]

To install additional .NET Core runtimes or SDKs:
  https://aka.ms/dotnet-download
Kevins-MBP:~ kevinr$  dotnet fsi

Welcome to .NET Core 3.1!
---------------------
SDK Version: 3.1.100-preview3-014645

Telemetry
---------
The .NET Core tools collect usage data in order to help us improve your experience. The data is anonymous. It is collected by Microsoft and shared with the community. You can opt-out of telemetry by setting the DOTNET_CLI_TELEMETRY_OPTOUT environment variable to '1' or 'true' using your favorite shell.

Read more about .NET Core CLI Tools telemetry: https://aka.ms/dotnet-cli-telemetry

----------------
Explore documentation: https://aka.ms/dotnet-docs
Report issues and find source on GitHub: https://github.com/dotnet/core
Find out what's new: https://aka.ms/dotnet-whats-new
Learn about the installed HTTPS developer cert: https://aka.ms/aspnet-core-https
Use 'dotnet --help' to see available commands or visit: https://aka.ms/dotnet-cli-docs
Write your first app: https://aka.ms/first-net-core-app
--------------------------------------------------------------------------------------

Microsoft (R) F# Interactive version 10.7.0.0 for F# 4.7
Copyright (c) Microsoft Corporation. All Rights Reserved.

For help type #help;;

> 2+2;;
val it : int = 4

> 
````

**Comment by cartermp** [comment]
Tagging as `needs-repro`, since although I can reproduce it that must not evidently be the case.

**Comment by KevinRansom** [comment]
We have a repro of this now.

**Comment by KevinRansom** [comment]
@cartermp , @brettfo this was fixed by this PR:  https://github.com/dotnet/fsharp/pull/7725

The commit with this change is in the release/dev16.5 branch.  However, the latest builds of dotnet cli has not been updated to include the most recent builds:

https://github.com/dotnet/toolset/blob/a5ba7188a56f57ec58df5080123561f749e9cb80/eng/Versions.props#L34

We need to follow up to ensure that master is flowing, or at least takes this fix.

Kevin

**Comment by mastoj** [comment]
@KevinRansom was this supposed to be fixed in SDK 3.1.100 released on December 3rd? I noticed that the PR was merged a while back so I would expect it to be, but I still see this bug: 

```
➜  ~ dotnet fsi

Microsoft (R) F# Interactive version 10.7.0.0 for F# 4.7
Copyright (c) Microsoft Corporation. All Rights Reserved.

For help type #help;;

> let x = 2 + 2;;

  let x = 2 + 2;;
  --------^

/Users/tomasjansson/stdin(1,9): error FS0074: The type referenced through 'System.Decimal' is defined in an assembly that is not referenced. You must add a reference to assembly 'The system type 'System.Decimal' was required but no referenced system DLL contained this type'.

> #quit;;
gcloud: uc-sdk-ssp  kube:  gke_uc-pure_us-central1-b_dinosaur  ⌚ 21:07:12
➜  ~ dotnet --version
3.1.100
```

Output from `dotnet --info`

```
➜  ~ dotnet --info
.NET Core SDK (reflecting any global.json):
 Version:   3.1.100
 Commit:    cd82f021f4

Runtime Environment:
 OS Name:     Mac OS X
 OS Version:  10.15
 OS Platform: Darwin
 RID:         osx.10.15-x64
 Base Path:   /usr/local/share/dotnet/sdk/3.1.100/

Host (useful for support):
  Version: 3.1.0
  Commit:  65f04fb6db

.NET Core SDKs installed:
  1.0.1 [/usr/local/share/dotnet/sdk]
  1.0.4 [/usr/local/share/dotnet/sdk]
  2.0.0 [/usr/local/share/dotnet/sdk]
  2.1.105 [/usr/local/share/dotnet/sdk]
  2.1.302 [/usr/local/share/dotnet/sdk]
  2.1.503 [/usr/local/share/dotnet/sdk]
  2.2.100 [/usr/local/share/dotnet/sdk]
  3.0.100-preview7-012821 [/usr/local/share/dotnet/sdk]
  3.0.100 [/usr/local/share/dotnet/sdk]
  3.1.100 [/usr/local/share/dotnet/sdk]

.NET Core runtimes installed:
  Microsoft.AspNetCore.All 2.1.2 [/usr/local/share/dotnet/shared/Microsoft.AspNetCore.All]
  Microsoft.AspNetCore.All 2.1.7 [/usr/local/share/dotnet/shared/Microsoft.AspNetCore.All]
  Microsoft.AspNetCore.All 2.2.0 [/usr/local/share/dotnet/shared/Microsoft.AspNetCore.All]
  Microsoft.AspNetCore.App 2.1.2 [/usr/local/share/dotnet/shared/Microsoft.AspNetCore.App]
  Microsoft.AspNetCore.App 2.1.7 [/usr/local/share/dotnet/shared/Microsoft.AspNetCore.App]
  Microsoft.AspNetCore.App 2.2.0 [/usr/local/share/dotnet/shared/Microsoft.AspNetCore.App]
  Microsoft.AspNetCore.App 3.0.0-preview7.19365.7 [/usr/local/share/dotnet/shared/Microsoft.AspNetCore.App]
  Microsoft.AspNetCore.App 3.0.0 [/usr/local/share/dotnet/shared/Microsoft.AspNetCore.App]
  Microsoft.AspNetCore.App 3.1.0 [/usr/local/share/dotnet/shared/Microsoft.AspNetCore.App]
  Microsoft.NETCore.App 1.0.4 [/usr/local/share/dotnet/shared/Microsoft.NETCore.App]
  Microsoft.NETCore.App 1.0.5 [/usr/local/share/dotnet/shared/Microsoft.NETCore.App]
  Microsoft.NETCore.App 1.1.1 [/usr/local/share/dotnet/shared/Microsoft.NETCore.App]
  Microsoft.NETCore.App 1.1.2 [/usr/local/share/dotnet/shared/Microsoft.NETCore.App]
  Microsoft.NETCore.App 2.0.0 [/usr/local/share/dotnet/shared/Microsoft.NETCore.App]
  Microsoft.NETCore.App 2.0.7 [/usr/local/share/dotnet/shared/Microsoft.NETCore.App]
  Microsoft.NETCore.App 2.1.2 [/usr/local/share/dotnet/shared/Microsoft.NETCore.App]
  Microsoft.NETCore.App 2.1.7 [/usr/local/share/dotnet/shared/Microsoft.NETCore.App]
  Microsoft.NETCore.App 2.2.0 [/usr/local/share/dotnet/shared/Microsoft.NETCore.App]
  Microsoft.NETCore.App 3.0.0-preview7-27912-14 [/usr/local/share/dotnet/shared/Microsoft.NETCore.App]
  Microsoft.NETCore.App 3.0.0 [/usr/local/share/dotnet/shared/Microsoft.NETCore.App]
  Microsoft.NETCore.App 3.1.0 [/usr/local/share/dotnet/shared/Microsoft.NETCore.App]

To install additional .NET Core runtimes or SDKs:
  https://aka.ms/dotnet-download
```

**Comment by cartermp** [comment]
@mastoj The fix is not in .NET Core 3.1. We'll need to see if we can service it.

I recommend removing all other SDKs on your machine. We noticed that with a clean machine and the .NET Core 3.1 SDK, it's fine.

**Comment by cartermp** [comment]
In the meantime, I highly recommend using Jupyter Notebooks instead of "raw" FSI in any non-production scenario. At this point I can confidently say it's superior in every way for interactive F# programming. You can learn how to set it up here: https://github.com/dotnet/try/blob/master/NotebooksLocalExperience.md

**Comment by mastoj** [comment]
@cartermp thanks. Deleted the other sdks and it seems to be working.

**Comment by cowlike** [comment]
I was having this same problem with version .NET Core 3.1 SDK on MacOS 10.14.6. I had many older SDKs installed and can confirm that removing them all and reinstalling 3.1 fixed it. I uninstalled using [this script](https://github.com/dotnet/cli/blob/master/scripts/obtain/uninstall/dotnet-uninstall-pkgs.sh)

**Comment by cartermp** [comment]
@cowlike nice find!

**Comment by KevinRansom** [comment]
@mastoj, the fix is for VS 16.5, it arrived too late for 16.4.  And there has been an issue with our code-flow into the 16.5 previews, so they don't yet have the fix.  @brettfo is tracking down the issue.

Kevin

**Comment by cartermp** [comment]
The fix that shipped with VS 16.5 and .NET Core SDK 3.1.201 should be live. I can also confirm that everything works in the latest .NET 5 preview. I'll close this out, but if folks still see it with SDK 3.1.201 or the latest .NET 5 preview then let us know.
